src/cleanupCommand/cleanupManager.ts

```diff
@@ -20,32 +22,48 @@ export class CleanupManager {
   ) {
     this.discreteBatchSize = config.get<number>('batch_size.discreteLayers');
     this.newIngestionJobType = config.get('new_ingestion_job_type');
+    this.updateIngestionJobType = config.get('update_ingestion_job_type');
+    this.sourceBlackList = config.get<string>('fs.blacklist_sources').split(',');
   }
 
   public async cleanFailedIngestionTasks(): Promise<void> {
     const FAILED_CLEANUP_DELAY = this.config.get<number>('failed_cleanup_delay_days.ingestion');
     const deleteDate = new Date();
     deleteDate.setDate(deleteDate.getDate() - FAILED_CLEANUP_DELAY);
-    const notCleanedAndFailed = await this.jobManager.getFailedAndNotCleanedIngestionJobs(this.newIngestionJobType);
 
-    for (let i = 0; i < notCleanedAndFailed.length; i += this.discreteBatchSize) {
-      const currentBatch = notCleanedAndFailed.slice(i, i + this.discreteBatchSize);
+    const notCleanedAndFailedNew = await this.jobManager.getFailedAndNotCleanedIngestionJobs(this.newIngestionJobType);
+    for (let i = 0; i < notCleanedAndFailedNew.length; i += this.discreteBatchSize) {
+      const currentBatch = notCleanedAndFailedNew.slice(i, i + this.discreteBatchSize);
       const expiredBatch = this.filterExpiredFailedTasks(currentBatch, deleteDate);
       if (expiredBatch.length > 0) {
-        await this.sourcesProvider.deleteDiscretes(expiredBatch);
+        const blackListFlitteredBatch = this.sourceBlackList.length > 0 ? this.filterBlackListSourcesTasks(expiredBatch) : expiredBatch;
```

Comment:
Spelling, rename to `blackListFiltteredBatch`.

Reply:
done